src/main/java/org/openrewrite/java/spring/boot3/RemoveConstructorBindingAnnotation.java

```diff
@@ -0,0 +1,88 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot3;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Alex Boyko
+ */
+public class RemoveConstructorBindingAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Remove Unnecessary @ConstructorBinding";
+    }
+
+    @Override
+    public String getDescription() {
+        return "As of Boot 3.0 @ConstructorBinding is no longer needed at the type level on @ConfigurationProperties classes and should be removed.";
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext context) {
+                int constrBindingAnnotationIndex = -1;
+                boolean isConfigPropsAnnotationPresent = false;
+                for (int i = 0; i < classDecl.getLeadingAnnotations().size() && (!isConfigPropsAnnotationPresent || constrBindingAnnotationIndex == -1); i++) {
+                    J.Annotation a = classDecl.getLeadingAnnotations().get(i);
+                    JavaType.FullyQualified fqType = TypeUtils.asFullyQualified(a.getType());
+                    if (fqType != null) {
+                        switch (fqType.getFullyQualifiedName()) {
+                            case "org.springframework.boot.context.properties.ConfigurationProperties":
+                                isConfigPropsAnnotationPresent = true;
+                                break;
+                            case "org.springframework.boot.context.properties.ConstructorBinding":
+                                constrBindingAnnotationIndex = i;
+                                break;
+                        }
+                    }
+                }
+                J.ClassDeclaration c = super.visitClassDeclaration(classDecl, context);
+                if (isConfigPropsAnnotationPresent && constrBindingAnnotationIndex >= 0) {
+                    return c.withLeadingAnnotations(removeAnnotation(c.getLeadingAnnotations(), constrBindingAnnotationIndex));
```

Comment:
Hi @BoykoAlex, have you considered using `ListUtils#map` for removing the annotation?  something like:
```java
if (isConfigPropsAnnotationPresent && constrBindingAnnotationIndex >= 0) {
    c = c.withLeadingAnnotations(ListUtils.map(c.getLeadingAnnotations(), anno -> {
        if (TypeUtils.isOfClassType(anno.getType(), "org.springframework.boot.context.properties.ConstructorBinding")) {
            maybeRemoveImport("org.springframework.boot.context.properties.ConstructorBinding");
            return null;
        }
        return anno;
    }));
}
```